src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -311,9 +308,10 @@ public void RemoveSubscriber_Impl(GuidId subscriptionId, StreamId streamId)
             var streamData = new StreamConsumerCollection();
             pubSubCache.Add(streamId, streamData);
             // Create a fake cursor to point into a cache.
-            // That way we will not purge the event from the cache to a potentially new alredy subscribed consumer, until we talk to pub sub.
-            // This will help ensure the "casual consistency" between pre-existing subsripton and later production.
-            var cursor = queueCache.GetCacheCursor(streamId.Guid, streamId.Namespace, firstToken);
+            // That way we will not purge the event from the cache, until we talk to pub sub.
+            // This will help ensure the "casual consistency" between pre-existing subscripton (of a potentially new already subscribed consumer) 
+            // and later production.
+            var fakeCursor = queueCache.GetCacheCursor(streamId.Guid, streamId.Namespace, firstToken);
```

Comment:
Nit: Not clear about the 'fakeCursor' name choice.  Maybe pinCuror to indicate it's pinning the data in the cache?  temporaryCursor, because we only have it around temporarily while we publish the stream?  Not sure of a better name, but 'fake' doesn't sound right.